examples/tests/valid/bypass_return.test.w

```diff
@@ -0,0 +1,28 @@
+test "bypass return type given throw is at parent level" {
+    let x: (): num = () => {
+        throw "not implemented";
+    };
+}
+
+test "bypass return when a throw is found" {
+    let x: (): num = () => {
+
+        // this should not be executed
+        if false {
+            return 1;
+        }
+
+        throw "not implemented";
+    };
+}
+
+test "return when a number is found even if a throw exists" {
+    let x: (): num = () => {
+
+        if true {
+            return 1;
+        }
+
+        throw "not implemented";
+    };
+}
```

Comment:
These tests are at the type-checker level, so no need to use tests.
Also, the last test isn't really different from the second one since we don't do any sort of data flow analysis.

```suggestion
let simpleThrow: (): num = () => {
  throw "not implemented";
};

let throwWithConditional: (): num = () => {
  // this should not be executed
  if false {
    return 1;
  }

  throw "not implemented";
};
```